veloroutes_voies_vertes/actions.py

```diff
@@ -110,9 +113,95 @@ def _remove_selected_from_relation(
     iface.messageBar().pushInfo('Véloroutes', msg)
 
 
-def remove_selected_portion(*args):
-    _remove_selected_from_relation(int(args[0]), 'id_itineraire', 'v_portion', 'id_portion', 'etape')
+def _add_selected_from_relation(
+    agg_id: int, src_primary_key: str, src_layer_name: str, rel_primary_key: str, rel_layer_name: str):
+
+    # get arguments
+    id_value = agg_id
+
+    # get project instance
+    project = QgsProject.instance()
 
+    # get layers needed
+    src_layer = project.mapLayersByName(src_layer_name)
+
+    # Control if layers exists
+    if not len(src_layer):
+        iface.messageBar().pushCritical('Véloroutes', 'La couche {} n\'a pas été trouvée'.format(
+            src_layer_name))
+        return
+    src_layer = src_layer[0]
+
+    rel_layer = project.mapLayersByName(rel_layer_name)
+    if not len(rel_layer):
+        iface.messageBar().pushCritical('Véloroutes', 'La table {} n\'a pas été trouvée'.format(
+            rel_layer_name))
+        return
+    rel_layer = rel_layer[0]
+
+    # count number of features selected from src_layer
+    count = src_layer.selectedFeatureCount()
+    if count < 1:
+        iface.messageBar().pushCritical(
+            'Véloroutes',
+            'Vous devez sélectionner au moins un objet de la couche {}'.format(src_layer_name))
+        return
 
-def remove_selected_segment(*args):
-    _remove_selected_from_relation(int(args[0]), 'id_portion', 'segment', 'id_segment', 'element')
+    # get list rel_layer
+    couple_rel = []
+    for feat in rel_layer.getFeatures(QgsExpression.createFieldEqualityExpression(src_primary_key, id_value)):
+        couple_rel.append((feat[src_primary_key], feat[rel_primary_key]))
+
+    # create list of couples between src_primary_key and rel_primary_key
+    features = src_layer.getSelectedFeatures()
+    couple_id = []
+    for feat in features:
+        couple_id.append((id_value, feat[rel_primary_key]))
+
+    # test between two lists
+    match_list = []
+    for item in couple_id:
+        if item not in couple_rel:
+            match_list.append(item)
+
+    # test if match_list is empty
+    if len(match_list) < 1:
+        iface.messageBar().pushInfo(
+            'Véloroutes',
+            'Pas d\'objet de la couche {} à ajouter'.format(rel_layer_name))
+        return
+
+    rel_layer.startEditing()
+    count = 0
+    for item in match_list:
+        feat = QgsFeature()
+        feat.setFields(rel_layer.fields())
+        feat[src_primary_key] = item[0]
+        feat[rel_primary_key] = item[1]
+        rel_layer.addFeatures([feat])
+        count += 1
+    rel_layer.commitChanges()
+    msg = "{} objet(s) ont été ajouté(s) de la couche {}".format(count, rel_layer_name)
```

Comment:
```suggestion
    msg = "{} objet(s) ont été ajouté(s) de la couche {}".format(len(match_list), rel_layer_name)
```
et supprimer la variable `count`?